### ISSUE TITLE
Query Fail using a presto-mysql connector. 

### ISSUE THREAD

**Issue by sumanth232** [facebook/presto#3054]
I am not able to do any query successfully on Mysql database using the presto-mysql connector.

```
$ cat etc/catalog/mysql.properties
connector.name=mysql
connection-url=jdbc:mysql://localhost:3306
connection-user=root
connection-password=asd
```

Here is the failure error in the console.

```
presto:default> SHOW SCHEMAS FROM mysql;

Query 20150608_141524_00022_j4z2k, FAILED, 1 node
Splits: 2 total, 0 done (0.00%)
0:00 [0 rows, 0B] [0 rows/s, 0B/s]

Query 20150608_141524_00022_j4z2k failed: null
```

The error stacktrace in IntelliJ

![screenshot 2015-06-08 15 25 41](https://cloud.githubusercontent.com/assets/4672368/8032322/c4f0606c-0df2-11e5-9877-2e253d3c5a05.png)

**Comment by sumanth232** [comment]
```
2015-06-08T18:10:15.985+0530	ERROR	SplitRunner-12-60	com.facebook.presto.execution.TaskExecutor	Error processing Split 20150608_124015_00003_j4z2k.1.0-0 (start = 1433767215966, wall = 19 ms, cpu = 0 ms, calls = 1)
java.lang.NullPointerException
	at com.facebook.presto.plugin.mysql.MySqlClient.getSchemaNames(MySqlClient.java:57)
	at com.facebook.presto.plugin.jdbc.JdbcMetadata.listSchemaNames(JdbcMetadata.java:61)
	at com.facebook.presto.metadata.MetadataManager.listSchemaNames(MetadataManager.java:206)
	at com.facebook.presto.connector.informationSchema.InformationSchemaPageSourceProvider.buildSchemata(InformationSchemaPageSourceProvider.java:241)
	at com.facebook.presto.connector.informationSchema.InformationSchemaPageSourceProvider.getInformationSchemaTable(InformationSchemaPageSourceProvider.java:130)
	at com.facebook.presto.connector.informationSchema.InformationSchemaPageSourceProvider.getInternalTable(InformationSchemaPageSourceProvider.java:115)
	at com.facebook.presto.connector.informationSchema.InformationSchemaPageSourceProvider.createPageSource(InformationSchemaPageSourceProvider.java:86)
	at com.facebook.presto.split.PageSourceManager.createPageSource(PageSourceManager.java:44)
	at com.facebook.presto.operator.TableScanOperator.createSourceIfNecessary(TableScanOperator.java:258)
	at com.facebook.presto.operator.TableScanOperator.isFinished(TableScanOperator.java:206)
	at com.facebook.presto.operator.Driver.processInternal(Driver.java:377)
	at com.facebook.presto.operator.Driver.processFor(Driver.java:303)
	at com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:586)
	at com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:505)
	at com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:639)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

```

**Comment by yuananf** [comment]
Can you connect to your mysql database using mysql client?
If so, can you connect to your mysql database using jdbc? You can write a simple java programe to confirm it.
In my opinion, 'localhost' is a bad choice for connection-url better be mysql database ip address, like '192.168.0.100'.

**Comment by sumanth232** [comment]
@yuananf , I can connect to mysql database using mysql client. I also wrote a java program to access the database using jdbc , and it works fine with all the below urls : 
connection-url=jdbc:mysql:localhost:3306/
connection-url=jdbc:mysql:localhost/
connection-url=jdbc:mysql:127.0.0.1:3306/
connection-url=jdbc:mysql:127.0.0.1/

I also added the mysql-connector-java-5.1.18-bin.jar file in presto IntelliJ project. I have no idea, why this simple thing is not working. Pls help ...

Do I need to make any other settings ?
